tokio/src/runtime/basic_scheduler.rs

```diff
@@ -266,63 +474,51 @@ impl SchedulerPriv {
         F::Output: Send + 'static,
     {
         let (task, handle) = task::joinable(future);
-        self.schedule_local(task);
+        self.queues.push_local(task);
         handle
     }
-
-    unsafe fn schedule_local(&self, task: Task<Self>) {
-        (*self.local_queue.get()).push_back(task);
-    }
-
-    fn next_task(&self, tick: u8) -> Option<Task<Self>> {
-        if 0 == tick % CHECK_REMOTE_INTERVAL {
-            self.next_remote_task().or_else(|| self.next_local_task())
-        } else {
-            self.next_local_task().or_else(|| self.next_remote_task())
-        }
-    }
-
-    fn next_local_task(&self) -> Option<Task<Self>> {
-        unsafe { (*self.local_queue.get()).pop_front() }
-    }
-
-    fn next_remote_task(&self) -> Option<Task<Self>> {
-        self.remote_queue.lock().unwrap().queue.pop_front()
-    }
 }
 
 impl Schedule for SchedulerPriv {
     fn bind(&self, task: &Task<Self>) {
         unsafe {
-            (*self.owned_tasks.get()).insert(task);
+            // safety: `Queues::add_task` is only safe to call from the thread
+            // that owns the queues (the thread the scheduler is running on).
+            // `Scheduler::bind` is called when polling a task that
+            // doesn't have a scheduler set. We will only poll tasks from the
+            // thread that the scheduler is running on. Therefore, this is safe
+            // to call.
+            self.queues.add_task(task);
```

Comment:
We don't have great terminology for this action. "bind" / "add_task" / ....

Nothing to do now, but maybe ponder how we can rename `bind` or describe it better.

Reply:
I thought about calling the method `Queues::bind` as well, but I thought it seemed a little weird to have a `Schedule::bind` that basically just made an assertion and then called `Queues::bind`.

FWIW, I am somewhat in favour of renaming the `Schedule` method for the reason that a search for `bind` in tokio also turns up a bunch of socket methods. Not a _big_ deal though.

Reply:
Also: documenting when and where the different `Schedule` methods are called would be really helpful — I think I have a good handle on it now, but it was confusing early on, and it would be nice to have something to refer to in reviews. I might try to do that in a follow-up.